02_bubble/jimbotech/bubble.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	sorted := make([]int, len(s))
+	copy(sorted, s)
+
+	var ss []int
+	for k := range sorted {
+		ss = sorted[:len(sorted)-k-1]
```

Comment:
Since `ss` is used only inside the loop, there's no point declaring the variable outside the loop. You can also use the `:=` syntax:
```
ss := sorted[:len(sorted)-k-1]
```

However, since you only use `ss` in one place too, you may as well just put it there without the variable:
```
for i, val := range sorted[:len(sorted)-k-1]
```

I was looking up and down the function to verify if `ss` was used anywhere else. Putting it in the `for...range` loop makes it clear it is not.

Reply:
Done